### PR TITLE
fix: createIndex attribute select changing all other selects

### PR DESCRIPTION
**Pull request by TGlide**
## What does this PR do?

- Fixes a bug in which changing the last Attribute select value would change all other Attribute select values. I also re-factored a bit of the code.
- Fixes a bug in which if you add a new attribute, and then delete it, you can't create an index

## Test Plan

Manual

## Related PRs and Issues

- https://github.com/appwrite/appwrite/issues/4762

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/indexes/createIndex.svelte [hunk 1]
```diff
@@ -171,16 +181,8 @@
...
-            noMargin
-            on:click={() => {
-                newAttr = true;
-                if (selectedAttribute && selectedOrder) {
-                    attributeList.push({ value: selectedAttribute, order: selectedOrder });
-                    selectedAttribute = selectedOrder = '';
-                }
-            }}>
+
+        <Button text noMargin on:click={addAttribute}>
```
Comment (ArmanNik) [inline src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/indexes/createIndex.svelte:185]: This button should be disabled when the new attribute `value` and `order` aren't selected (right now is active but does nothing when clicked)

File: package-lock.json [hunk 2]
```diff
@@ -6882,9 +6882,9 @@
             }
         },
         "node_modules/svelte": {
-            "version": "3.52.0",
-            "resolved": "https://registry.npmjs.org/svelte/-/svelte-3.52.0.tgz",
-            "integrity": "sha512-FxcnEUOAVfr10vDU5dVgJN19IvqeHQCS1zfe8vayTfis9A2t5Fhx+JDe5uv/C3j//bB1umpLJ6quhgs9xyUbCQ==",
+            "version": "3.53.1",
+            "resolved": "https://registry.npmjs.org/svelte/-/svelte-3.53.1.tgz",
+            "integrity": "sha512-Q4/hHkktZogGhN5iqxqSi9sjEVoe/NbIxX4hXEHoasTxj+TxEQVAq66LnDMdAZxjmsodkoI5F3slqsS68U7FNw==",
```
[the remark this answers is not in the document]
Comment (TGlide) [reply package-lock.json:6887]: I'll revert this then!

File: src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/indexes/createIndex.svelte [hunk 3]
```diff
@@ -47,91 +47,103 @@
...
-                });
-                trackEvent('submit_index_create');
-            } catch (error) {
-                addNotification({
-                    message: error.message,
-                    type: 'error'
-                });
-            }
 
+    const create = async () => {
```
[the remark this answers is not in the document]
Comment (TGlide) [reply src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/indexes/createIndex.svelte:51]: I'll change it now 🙂

File: src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/indexes/createIndex.svelte [hunk 4]
```diff
@@ -141,49 +104,41 @@
...
-                    selectedAttribute = selectedOrder = '';
-                }
-            }}>
+        {/each}
+
+        <Button text noMargin on:click={addAttribute} disabled={addAttributeDisabled}>
             <span class="icon-plus" aria-hidden="true" />
             <span class="text">Add attribute</span>
         </Button>
+        {JSON.stringify(attributeList)}
```
Comment (stnguyen90) [inline src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/indexes/createIndex.svelte:141]: Please make sure to remove this before making the PR ready for review! 🙏🏼

**Review by stnguyen90 — APPROVED** [review]

**Review by TorstenDittmann — APPROVED** [review]

**Review by ArmanNik — APPROVED** [review]

**Pull request merged**